### PR TITLE
fix: proper error handling for snyk test command

### PR DESCRIPTION
**Pull request by kyegupov**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Fixes the error handling so that we don't get the mysterious "Failed to get vulns" message for incorrect auth.

Drive-by: typescriptify fake-server

### CODE REVIEW

File: src/lib/snyk-test/run-test.ts [hunk 1]
```diff
@@ -183,9 +184,15 @@ function handleTestHttpErrorResponse(res, body) {
   let err;
   const userMessage = body && body.userMessage;
   switch (statusCode) {
-    case (statusCode === 500):
+    case 401:
+    case 403:
```
Comment (lwywoo) [inline src/lib/snyk-test/run-test.ts]: Please can a test be added for 401/403 cases

File: src/lib/snyk-test/run-test.ts [hunk 2]
```diff
@@ -183,9 +184,15 @@ function handleTestHttpErrorResponse(res, body) {
...
+      err = AuthFailedError(userMessage, statusCode);
+      err.innerError = body.stack;
+      break
```
Comment (lwywoo) [inline src/lib/snyk-test/run-test.ts]: Missing semi colon

**Comment by lili2311** [comment]
Please add tests for these errors? Do existing tests cover it already?

File: src/lib/snyk-test/run-test.ts (hunk 1 above)
Comment (kyegupov) [reply src/lib/snyk-test/run-test.ts]: Added.

File: src/lib/snyk-test/run-test.ts (hunk 2 above)
Comment (kyegupov) [reply src/lib/snyk-test/run-test.ts]: Fixed.

**Comment by kyegupov** [comment]
@lili2311 nope, weren't covered. Added.

**Pull request merged**